pkg/server/metrics/metrics.go

```diff
@@ -187,3 +215,17 @@ func (a *ServerMetrics) SetPendingDialCount(count int) {
 func (a *ServerMetrics) FullRecvChannel(serviceMethod string) prometheus.Gauge {
 	return a.fullRecvChannels.With(prometheus.Labels{"service_method": serviceMethod})
 }
+
+type DialFailureReason string
+
+const (
+	DialFailureEndpoint             DialFailureReason = "endpoint"              // Dial failure reported by the agent back to the server.
+	DialFailureUnrecognizedResponse DialFailureReason = "unrecognized_response" // Dial repsonse received for unrecognozide dial ID.
+	DialFailureSend                 DialFailureReason = "send"                  // Successful dial response from agent, but failed to send to frontend.
```

Comment:
Thinking about how specific we want to be with the reasons. For e.g. would `DialFailureSendResponseToFrontend` make things clearer or just adding too much verbosity?

Reply:
How about `DialFailureSendResponse = "send_rsp"` as a compromise? The server shouldn't ever be sending a DIAL_RSP anywhere else, so I think frontend is redundant here?

Reply:
sgtm